### PR TITLE
decouples outputenable state and wl_output global

### PR DESCRIPTION
**Pull request by Ongy**
This decouples wlr_output_enable and the wl_global.
The previously internal functions wlr_output_(destroy/create)_global are
exposed and used automatically in the wlr_output_layout to create/tear
down the global.
The compositor can handle them itself if it wants to, but I think this
is the right moment to create/destroy the wl_output when the
wlr_output_layout is used.

Examples are not updates as they either use `wlr_output_layout` or won't make use of the `wl_output` either way.

### CODE REVIEW

**Comment by Ongy** [comment]
Forgot to mention, this is for https://github.com/swaywm/wlroots/issues/568

The one thing I'd consider as well is to set outputs into a disabled state by default. But since being disabled is currently a drm specific thing, I think it's better to start in the mode all backends handle well.

**Comment by SirCmpwn** [comment]
>The one thing I'd consider as well is to set outputs into a disabled state by default.

Hmm, probably not. But I don't feel strongly about it.

File: include/wlr/types/wlr_output.h [hunk 1]
```diff
@@ -89,6 +89,8 @@ struct wlr_output {
 struct wlr_surface;
 
 void wlr_output_enable(struct wlr_output *output, bool enable);
+void wlr_output_create_global();
+void wlr_output_destroy_global();
```
Comment (emersion) [inline include/wlr/types/wlr_output.h]: This is missing the `output` parameter.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**